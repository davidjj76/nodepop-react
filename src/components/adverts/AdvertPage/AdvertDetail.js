import React from 'react';
import T from 'prop-types';

import { Button, ConfirmationButton } from '../../shared';
import placeholder from '../../../assets/images/placeholder.png';
import { advert } from '../propTypes';

function AdvertDetail({ name, sale, price, tags, photo, onDelete }) {
  return (
    <div>
      <p>{name}</p>
      <p>{sale ? 'Sell' : 'Buy'}</p>
      <p>{tags.join(', ')}</p>
      <p>{price}</p>
      <img
        src={photo || placeholder}
        alt={name}
        width="200"
        height="200"
        style={{ objectFit: 'contain' }}
      />
      <ConfirmationButton
        variant={Button.variants.danger}
        confirmation="Are you sure you want to delete this advert?"
        onConfirm={onDelete}
      >
        Delete
      </ConfirmationButton>
    </div>
  );
}

AdvertDetail.propTypes = {
  ...advert,
  photo: T.string,
  onDelete: T.func.isRequired,
};

AdvertDetail.defaultProps = {
  photo: null,
};

export default AdvertDetail;
