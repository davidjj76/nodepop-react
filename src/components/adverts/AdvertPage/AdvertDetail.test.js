import { shallow } from 'enzyme';
import AdvertDetail from './AdvertDetail';

describe('AdvertDetail', () => {
  let wrapper;
  const props = {
    name: 'Coche',
    sale: true,
    tags: ['motor'],
    price: 5000,
    onDelete: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<AdvertDetail {...props} />);
  });

  test('should render snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call onDelete', () => {
    wrapper.find('ConfirmationButton').props().onConfirm();
    expect(props.onDelete).toHaveBeenCalled();
  });
});
