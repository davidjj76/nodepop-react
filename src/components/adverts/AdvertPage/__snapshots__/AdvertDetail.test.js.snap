// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvertDetail should render snapshot 1`] = `
<div>
  <p>
    Coche
  </p>
  <p>
    Sell
  </p>
  <p>
    motor
  </p>
  <p>
    5000
  </p>
  <img
    alt="Coche"
    height="200"
    src="placeholder.png"
    style={
      Object {
        "objectFit": "contain",
      }
    }
    width="200"
  />
  <ConfirmationButton
    confirmation="Are you sure?"
    onConfirm={[MockFunction]}
  >
    Delete
  </ConfirmationButton>
</div>
`;
